tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/generators/cpp/CppClientGenerator.java

```diff
@@ -208,6 +208,37 @@ protected void addRequestIdToResults(final ServiceModel serviceModel) {
                 });
     }
 
+    private static Set<String> servicesMissingMultiAuthMRAPTrait = new HashSet<>();
+    static {
+        servicesMissingMultiAuthMRAPTrait.add("S3");
+        servicesMissingMultiAuthMRAPTrait.add("S3-CRT");
+        servicesMissingMultiAuthMRAPTrait.add("CloudFront KeyValueStore");
+        servicesMissingMultiAuthMRAPTrait.add("SESv2");
+        servicesMissingMultiAuthMRAPTrait.add("EventBridge");
+    }
+
+    private void CheckAndEnableSigV4A(final ServiceModel serviceModel, VelocityContext context) {
+        List<String> c2jAuthList = serviceModel.getMetadata().getAuth();
+        String serviceId = serviceModel.getMetadata().getServiceId();
+        if (c2jAuthList != null && c2jAuthList.contains("aws.auth#sigv4a") ||
+             servicesMissingMultiAuthMRAPTrait.contains(serviceId)) {
+            context.put("multiRegionAccessPointSupported", true);
+        }
+        // todo: remove these checks later
+        if (serviceModel.getEndpointRules().contains("\"sigv4a\"") &&
```

Comment:
besides endpoint rules, I think we should also check in operations. at operation level there can be authscheme in "auth" field (example cognito idp)

Reply:
updated, thank you.
although, I'm not sure we need to check for noauth or bearer per operation.
This block is for enabling MRAP flag only, not for c2j model validation.